### DISCUSSION TITLE
Rauchwarnmelder HmIP-SWSD - Can not trigger an action based on "Primary alarm"

### DISCUSSION THREAD

**Discussion by hklages** [SukramJ/aiohomematic#774] [Q&A] [answered]
Hi. I would like to send a notification to my smartphone in case smoke is being detected. As trigger best would be to "Primary alarm", because that will cause only one trigger.  

When creating the automation I am not able to choose "Primary alarm" or "Secondary alarm" - only "unavailable"/ "unknown" is shown.

![States](https://user-images.githubusercontent.com/17273119/209160171-1b8f50f3-56c2-4dc9-baf7-5b24d1bdd9e6.jpg)

If I enter the text ("Primary alarm" or even Secondary alarm manually the automation is never triggered. If I leave it blank I get  6 notifications (2 for each sensor), when changing to "Primary alarm"/"Secondary alarm" and back to idle. 

The log file shows the state correctly. 
![smoke detector log](https://user-images.githubusercontent.com/17273119/209161194-70ecc906-21cc-4152-a30e-48d70c5b0efc.jpg)

Do I do something wrong?

**Comment by SukramJ** [comment]
Don't use `device` as trigger try `state`

**Comment by hklages** [comment]
Thats what I do - States, using an entity

![state](https://user-images.githubusercontent.com/17273119/209166620-1d4b2033-074a-49f7-bfce-ab0dff9e2abb.jpg)

**Reply by SukramJ** [reply]
Don't know why there are 2 Entity lines.
Add `To` with the target status

**Comment by hklages** [comment]
I added target status "Primary alarm" and even "Secondary alarm" but that does not work. 

The following script does not trigger any event when the state changes to "Primary alarm". 
As you can see, trigger platform is state and to is "Primary alarm" (as target)

`alias: 111 On Smoke Detected Do Alarm Notification
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.smoke_detector_bed_room_smoke_detector_alarm_status
      - sensor.smoke_detector_entrance_smoke_detector_alarm_status
      - sensor.smoke_detector_office_smoke_detector_alarm_status
    to: Primary alarm
  - platform: state
    entity_id:
      - sensor.smoke_detector_bed_room_smoke_detector_alarm_status
      - sensor.smoke_detector_entrance_smoke_detector_alarm_status
      - sensor.smoke_detector_office_smoke_detector_alarm_status
    to: Secondary alarm
condition: []
action:
  - service: script.notification_to_smartphone
    data:
      notification_title: ALARM
      notification_message: Smoke detected via HmIP-SWSD!
mode: single`

**Reply by SukramJ** [reply]
Should be `primary_alarm`, `intrusion_alarm`, `secondary_alarm` or `idle_off`

```
    "homematicip_local__smoke_detector_alarm_status": {
      "idle_off": "Untätig",
      "primary_alarm": "Primär-Alarm",
      "intrusion_alarm": "Einbruch-Alarm",
      "secondary_alarm": "Sekundär-Alarm"
    },
```

**Reply by SukramJ** [reply]
Next release will have an additional attribute `value_list` with the correct values for enum-style sensor entities.

**Comment by hklages** [comment, marked as answer]
Great. I see the value_list in developer tools. When using primary_alarm and secondary_alarm the trigger works and I receive a message on my smartphone (and the cover opens, lights are turned on). 

Thanks!

**Reply by SukramJ** [reply]
Next change will be in january:
HA adds options to the sensor model. That makes the value selectable in the UI.
